### ISSUE TITLE
Unable to set static ip address using vagrant-vshpere

### ISSUE THREAD

**Issue by VarunUmesh** [nsidc/vagrant-vsphere#80]
I am unable to set static IP address using this plugin. I have the template in vSphere [a vCenter template] and it is a SUSE linux . The VM boots up properly, but IP address does not gets correctly assigned.

The contents of my Vagrantfile looks something like this:

Vagrant.configure("2") do |config|
  config.vm.box = 'dummy'
  config.vm.box_url = './example_box/dummy.box'

  config.vm.provider :vsphere do |vsphere , override|
    override.vm.network 'private_network' , ip: 'xxx.xxx.xxx.xxx'
    vsphere.customization_spec_name = '<My_Customization_Specification_Manager_In_vSphere>'
  end

  config.vm.provider :vsphere do | vsphere |
    vsphere.host = 'xxx.xxx.xxx.xxx'
    vsphere.data_center_name = 'My_DataCenter_Name'
    vsphere.data_store_name = 'My_Store_Name'
    vsphere.template_name = 'Template_Name_In_vSphere'
    vsphere.name = 'Name_Of_The_VM'
    vsphere.user = '<User Name>'
    vsphere.password = '<Password>'
    vsphere.insecure = true
    vsphere.compute_resource_name = 'xxx.xxx.xxx.xxx'
  end

I have also made the necessary settings in the Customization Specification Manager part in vShpere. When i see the virtual machine in the console, it fails to bring up the eth0 on the machine.

**Comment by dylancwood** [comment]
I've had some success with this when bringing up Ubuntu12.04 guest OS. It is necessary to specify a customization spec in the Vagrantfile:

`vsphere.customization_spec_name = 'my-spec'`

Where `my-spec` refers to a customization spec created in vsphere (see https://pubs.vmware.com/vsphere-51/index.jsp?topic=%2Fcom.vmware.vsphere.vm_admin.doc%2FGUID-C72FA9A6-3871-4838-A4AE-50E218488CD4.html)

From what I've seen, it is necessary to specify the desired IP in the customization spec (the 'prompt user for an address' option results in `undefined method ipAddress=' for #<RbVmomi::VIM::CustomizationUnknownIpGenerator:0x000000026970b0>
`). I find this rather inconvenient, as I am hoping to consolidate all of my VM creation config into the Vagrantfile, and making a separate customization-spec in vsphere for each VM I manage goes against that.

**Anyone have any idea how to create a single customization spec that will allow vagrant to specify the IP?**

**Comment by dylancwood** [comment]
Update, it appears that the IP address specified in the Vagrantfile will override whatever IP is specified in the VMWare customization-spec.

**Comment by pforai** [comment]
Same issue here - any plans on fixing this?

**Comment by jwbraucher** [comment]
I think setting a static IP address with this solution only works if vSphere supports guest OS customization for that particular OS. There is a support matrix on vmware.com with a list of guest OS's and vSphere releases that work together with customization specs:
 http://partnerweb.vmware.com/programs/guestOS/guest-os-customization-matrix.pdf

It sounds like @dylancwood is on the right track. I would highly suggest reading through the vSphere documentation he linked (above) about setting up customization specs and perhaps testing your deployments through the vSphere client before attempting them with vagrant-vsphere.

To my knowledge, vagrant-vsphere doesn't do any network customization itself. It just overrides the IP address used by vSphere during its guest OS customization process.

**Comment by toddc-unity3d** [comment]
Agreed, it looks like the guest OS has to support that.  It may be ugly but something that works for me is to write a wrapper script that in addition to firing up a machine SSH's in and reconfigures that part, THEN runs Puppet or anything else I need.  It's not pretty, but on OSes that don't support customization you're kinda stuck.

**Comment by VarunUmesh** [comment]
Okay, another work around would be to have a DHCP server installed which would give the IP addresses when you clone the virtual machine and then do a vagrant provisioning [https://docs.vagrantup.com/v2/provisioning/shell.html], which allows you to make configurations after the vagrant up command. Thus you would be using your DHCP only temporarily and then assign the required IP after boot up

**Comment by rylarson** [comment]
@jwbraucher is correct, setting a static IP using `:private_network` and specifying a customization spec requires that vsphere supports guest OS customization for your OS version. The VM must also have vmware tools installed for guest customization to work, and minimum VM version requirements apply to.

If you are having problems with this feature, please state:

   * Your vCenter version (including update level)
   * OS Version
   * vmware tools version
   * vm version

**Comment by karnold** [comment]
I am also experiencing the same issue with the following.

vSphere 6.0
Ubuntu 14.04 as the VM OS
Mac OS X 10.11 as the vagrant workstation
open-vm-tools 9.4.0

**Comment by rylarson** [comment]
@karnold What VM hardware version is your guest OS?

**Comment by karnold** [comment]
VM Hardware version 11

**Comment by simsunny22** [comment]
so I have same problem
I have also made the necessary settings in the Customization Specification Manager part in vShpere. When i see the virtual machine in the console, it fails to bring up the eth0 on the machine.
and i can bring up eth0 by ifup etho command.
is some thing wrong with my Customization Specification.
thank you for any help

**Comment by doker78** [comment]
problem exist -  does anyone planning to  check it ?

**Comment by suikast42** [comment]
I run into same issue with Centos8 base image. 

After one day research I found a working workflow.

1. Install a Centos 8 image from a ISO. During the install choose a custom IP setting ( not DHCP) 
2. Asfter the machine is setted up login via ssh and execute the folowing commands
```console
dnf update -y
dnf clean all
systemctl start vmtoolsd.service
systemctl enable vmtoolsd.service
rm -vf /etc/ssh/ssh_host_*
dnf install -y perl
systemctl stop systemd-journald.socket
find /var/log -type f -exec rm {} \;
mkdir -p /var/log/journal
cd /root ; rm -f .bash_history ; history -c
systemctl poweroff
```
3. Convert the machine to a template
4. Create a VM Customization Specifications with a static IP.
